### PR TITLE
BAQE-380: Increase coverage of guided rule attributes widgets

### PR DESCRIPTION
**Pull request by jomarko**
This PR extracts rule attributes widgests logic into separate classes to facilitate testing.
The testing introduced in the PR focuses just on some attributes.

### CODE REVIEW

**Comment by jomarko** [comment]
@manstis I will be pleased with your review. If you have a look on [BAQE-380](https://issues.jboss.org/browse/BAQE-380) this PR is corollary of the effort decreasing amount of selenium tests.

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/widget/attribute/EditAttributeWidgetFactory.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import org.drools.workbench.models.datamodel.rule.RuleAttribute;
+import org.gwtbootstrap3.client.ui.TextBox;
+import org.kie.workbench.common.widgets.client.widget.TextBoxFactory;
+
+public class EditAttributeWidgetFactory {
+
+    final boolean isReadOnly;
+
+    public EditAttributeWidgetFactory(boolean isReadOnly) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/widget/attribute/EditAttributeWidgetFactory.java]: I know you're one to comment on use of ```final```... want to add here?

Comment (manstis) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/widget/attribute/EditAttributeWidgetFactory.java]: @jomarko Looks like you add it in [this](https://github.com/kiegroup/drools-wb/pull/866) PR, so no need to worry on this one.

Comment (jomarko) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/widget/attribute/EditAttributeWidgetFactory.java]: With pleasure!

**Pull request merged**